### ISSUE TITLE
Device Nodes breaking after device restart

### ISSUE THREAD

**Issue by Tobiti** [zachowj/node-red-contrib-home-assistant-websocket#529]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

When I restart my raspberry pi the device nodes are not getting any event from home assistant. When I now redeploy the flow, its working again.

### To Reproduce

1. Restart your device
2. No device node is working for me

### Expected behavior

A restart is not effecting the node behavior

### Example Flow

```json
[
    {
        "id": "90293ea70231a584",
        "type": "ha-device",
        "z": "ee9814b494a333b6",
        "g": "ddd6d0148444ebcb",
        "name": "Enable Button Tapped",
        "server": "c42bb865d28d84c7",
        "version": 0,
        "exposeToHomeAssistant": false,
        "haConfig": [
            {
                "property": "name",
                "value": ""
            },
            {
                "property": "icon",
                "value": ""
            }
        ],
        "inputs": 0,
        "deviceType": "trigger",
        "device": "be2903067ad98c2117726db460a00e5f",
        "event": {
            "device_id": "be2903067ad98c2117726db460a00e5f",
            "domain": "zha",
            "platform": "device",
            "type": "remote_button_short_press",
            "subtype": "turn_on"
        },
        "capabilities": [],
        "outputProperties": [
            {
                "property": "payload",
                "propertyType": "msg",
                "value": "",
                "valueType": "eventData"
            },
            {
                "property": "topic",
                "propertyType": "msg",
                "value": "",
                "valueType": "triggerId"
            }
        ],
        "x": 140,
        "y": 60,
        "wires": [
            [
                "656b5a2027d8d460"
            ]
        ]
    },
    {
        "id": "656b5a2027d8d460",
        "type": "api-call-service",
        "z": "ee9814b494a333b6",
        "g": "ddd6d0148444ebcb",
        "name": "Turn light on",
        "server": "c42bb865d28d84c7",
        "version": 3,
        "debugenabled": false,
        "service_domain": "light",
        "service": "turn_on",
        "entityId": "light.pc_lampe_level_light_color_on_off",
        "data": "",
        "dataType": "jsonata",
        "mergecontext": "",
        "mustacheAltTags": false,
        "outputProperties": [],
        "queue": "none",
        "x": 390,
        "y": 60,
        "wires": [
            []
        ]
    },
    {
        "id": "c42bb865d28d84c7",
        "type": "server",
        "name": "Home Assistant",
        "version": 2,
        "addon": false,
        "rejectUnauthorizedCerts": true,
        "ha_boolean": "y|yes|true|on|home|open",
        "connectionDelay": true,
        "cacheJson": true,
        "heartbeat": false,
        "heartbeatInterval": "30"
    }
]
```

### This package's version

0.39.0

### Is Node-RED running in Docker?

Yes

### Node-RED version

2.1.4

### Node.js version

Official nodered/node-red Docker

### Additional context

I think this is the error from home assistant:
Logger: custom_components.nodered.switch
Source: helpers/trigger.py:93
Integration: Node-RED Companion (documentation, issues)
First occurred: 22:46:06 (8 occurrences)
Last logged: 22:46:06

Unknown error while setting up trigger (empty result)

**Comment by erwineverts** [comment]
I have the same problem. I am running home assistant behind a traefik reverse proxy. Do you use a reverse proxy?

**Comment by Tobiti** [comment]
No I have both running in a docker and node red is using the local ip to connect to  home assistant

**Comment by Xeevis** [comment]
Same issue here. Both HA and Node-RED run in separate docker containers. If Node-RED is started before HA or if HA is restarted, device nodes stop triggering even though their state is "running". State nodes work fine though. Only way to fix device nodes is to deploy or restart flows.

**Comment by Xeevis** [comment]
As a temp fix Home Assistant when started can perform automation to start a flow which in turn can perform http request to Node-RED API to restart flows automatically.
https://zachowj.github.io/node-red-contrib-home-assistant-websocket/cookbook/starting-flow-after-home-assistant-restart.html

```js
msg.url = 'http://localhost:1880/flows';
msg.method = 'POST';
msg.headers = {};
msg.headers['content-type'] = "application/json; charset=utf-8";
msg.headers['Node-RED-Deployment-Type'] = "reload";
msg.headers['Node-RED-API-Version'] =  "v2";
/* msg.headers['Authorization'] = "Bearer <token>" */
msg.payload = {
    "flows": [
    ]
};
return msg;
```

**Comment by zachowj** [comment]
Currently, it waits for the NR integration to be loaded before attempting to register a device trigger. If I change it to wait for Home Assistant to be in a "running" state and then register the trigger it seems to work. I want to dig into it a little more to see what integration needs to be loaded before you can register a device trigger.

**Comment by zachowj** [comment]
Fixed in v1.0.1 of the custom component 

https://github.com/zachowj/hass-node-red/releases/tag/v1.0.1
